src/Domain/OrientationMap.hpp

```diff
@@ -85,3 +82,67 @@ bool operator!=(const OrientationMap<VolumeDim>& lhs,
                 const OrientationMap<VolumeDim>& rhs) noexcept {
   return not(lhs == rhs);
 }
+
+// `OrientationMap`s define an active rotation of the logical axes that bring
+// the axes of a host block into alignment with the logical axes of the
+// neighbor block. `DiscreteRotation` applies this active rotation on the
+// coordinates as opposed to the axes.
+// For a two-dimensional example, consider a host block and a neighbor block,
+// where the OrientationMap between them is \f$\{-\eta,+\xi\}\f$. A quarter-
+// turn counter-clockwise of the host block's logical axes would bring them into
+// alignment with those of the neighbor. That is, after this active rotation,
+// the blocks would be Aligned. Now consider a point A with coordinates
+// (+1.0,-0.5). An active quarter-turn rotation counter-clockwise of point A,
+// keeping the axes fixed, brings point A into the coordinates (+0.5,+1.0).
+// This is how `DiscreteRotation` interprets the `OrientationMap` passed to it.
+template <typename T>
+std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 1> DiscreteRotation(
+    const OrientationMap<1>& rotation,
+    const std::array<T, 1>& source_coords) noexcept {
+  using ReturnType = std::decay_t<tt::remove_reference_wrapper_t<T>>;
+  return std::array<ReturnType, 1>{{std::move(
+      source_coords[0] *
+      (rotation(Direction<1>::upper_xi()).side() == Side::Upper ? 1.0
+                                                                : -1.0))}};
+}
+
+template <typename T>
+std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 2> DiscreteRotation(
+    const OrientationMap<2>& rotation,
+    const std::array<T, 2>& source_coords) noexcept {
+  using ReturnType = std::decay_t<tt::remove_reference_wrapper_t<T>>;
+  return std::array<ReturnType, 2>{
+      {std::move(gsl::at(source_coords,
+                         rotation(Direction<2>::upper_xi()).dimension()) *
+                 (rotation(Direction<2>::upper_xi()).side() == Side::Upper
+                      ? 1.0
+                      : -1.0)),
+       std::move(gsl::at(source_coords,
+                         rotation(Direction<2>::upper_eta()).dimension()) *
+                 (rotation(Direction<2>::upper_eta()).side() == Side::Upper
+                      ? 1.0
+                      : -1.0))}};
+}
+
+template <typename T>
+std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3> DiscreteRotation(
+    const OrientationMap<3>& rotation,
+    const std::array<T, 3>& source_coords) noexcept {
+  using ReturnType = std::decay_t<tt::remove_reference_wrapper_t<T>>;
+  return std::array<ReturnType, 3>{
+      {std::move(gsl::at(source_coords,
+                         rotation(Direction<3>::upper_xi()).dimension()) *
```

Comment:
for loop, *= for -1 case only, move result, use enums for directions, collapse three functions into one, use if/elses